TLM/TLM/Manager/Impl/VehicleBehaviorManager.cs

```diff
@@ -334,6 +335,47 @@ public class VehicleBehaviorManager : AbstractCustomManager, IVehicleBehaviorMan
 							vehicleData.m_waitCounter = 0;
 						}
 
+                        // Check if turning in the preferred direction, and if turning while it's red is allowed
+                        if (Options.turnOnRed) {
+                            if (JunctionRestrictionsManager.Instance.IsTurnOnRedAllowed(prevPos.m_segment, isTargetStartNode)) {
+                                // Check if vehicle has stopped
+                                // TODO: Disallow cutting into flowing traffic, disrupting the flow.
+                                if (
+                                    (vehicleState.JunctionTransitState == VehicleJunctionTransitState.Stop && sqrVelocity <= TrafficPriorityManager.MAX_SQR_STOP_VELOCITY)
+                                    || vehicleState.recklessDriver && sqrVelocity <= TrafficPriorityManager.MAX_YIELD_VELOCITY) {
+
+                                    ushort uCurrentSegment = prevPos.m_segment;
+                                    ushort uTargetSegment = position.m_segment;
+
+                                    // If you can turn preferred and you're on a one-way road and not going straight, continue
+                                    SegmentGeometry currentSegGeo = SegmentGeometry.Get(uCurrentSegment);
+                                    if ((Constants.ServiceFactory.SimulationService.LeftHandDrive ? currentSegGeo.HasLeftSegment(true) : currentSegGeo.HasRightSegment(true)) ||
+                                        (currentSegGeo.IsOneWay() && !currentSegGeo.IsStraightSegment(uTargetSegment, false) && !currentSegGeo.IsStraightSegment(uTargetSegment, true))) {
+
+                                        ushort uTurnSegment = 0;
+
+                                        Constants.ServiceFactory.NetService.ProcessSegment(uCurrentSegment, delegate (ushort sId, ref NetSegment segment) {
+                                            if (Constants.ServiceFactory.SimulationService.LeftHandDrive) {
+                                                uTurnSegment = segment.GetLeftSegment(targetNodeId);
+                                            } else {
+                                                uTurnSegment = segment.GetRightSegment(targetNodeId);
+                                            }
+                                            return true;
+                                        });
+
+                                        // If the direction you are going is the same as the preferred lane or you're going from a one-way road to another one-way road, turn
+                                        SegmentGeometry turnSegGeo = SegmentGeometry.Get(uTurnSegment);
+                                        if (uTargetSegment == uTurnSegment || (currentSegGeo.IsOneWay() == true && turnSegGeo?.IsOneWay() == true)) {
```

Comment:
I think you can omit that first comparator with 'true' 😉 
 ```cs
if (uTargetSegment == uTurnSegment || (currentSegGeo.IsOneWay() && turnSegGeo?.IsOneWay() == true)) {
```

Reply:
Wow, I didn't even notice that. Good for readability 👍